### PR TITLE
Robots and sitemap doc

### PR DESCRIPTION
**Pull request by ngilain**

### CODE REVIEW

File: en/developer_guide/robots.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+matching textarea: the content of the ``robots.txt`` file will be exactly what is written here.
+
+.. _files generation:
+
+Files generation
+----------------
+Once contributed, the content of the ``robots.txt`` is only known from the database. To serve it to
+a client requesting it, two approaches are possible: use a controller action looking in the database
+for the information matching the current site, or serve a previously generated file containing these
+information. To enhance perfomances, Open Orchestra choose the second option. This way, there is no
```
Comment (nicolasThal) [inline en/developer_guide/robots.rst]: you could split this paragraph before `To enhance`

Comment (nicolasThal) [inline en/developer_guide/robots.rst]: This solution prevent any php action nor database access

File: en/developer_guide/robots.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+intervention of Symfony, PHP and no access to the Datatbase. These static files can also be easily
```
Comment (nicolasThal) [inline en/developer_guide/robots.rst]: As those file are static, they can easily be cached by a reverse proxy

File: en/developer_guide/robots.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+cached by a reverse proxy.
+
+So when ``robots.txt`` files are contributed, you have to dump them into physical files. To help you
```
Comment (nicolasThal) [inline en/developer_guide/robots.rst]: Once a `robots.txt` data is contributed, you have to manually dump it into a file. Open Orchestra provides a console command, available on your Front installation, that can be used.

File: en/developer_guide/robots.rst [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+to generate them, Open Orchestra provides a console command that must be used on your Front
+installation:
+
+    app/console orchestra:robots:generate [--siteId=SITEID]
+
+With that command, the ``robots.txt`` files of all sites are dumped, or only one if you provide a
```
Comment (nicolasThal) [inline en/developer_guide/robots.rst]: This command will dump one or all the robots.txt files, depending on the siteId parameter presence.

File: en/developer_guide/robots.rst [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+siteId. The generated files are stored in the web directory of your application, to allow them to
```
Comment (nicolasThal) [inline en/developer_guide/robots.rst]: To be accessible by web client, the generated files ...

File: en/developer_guide/robots.rst [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+be accessed by clients. Each file is stored in a sub-directory named by the site Id. For instance
+if you have three sites whose id is 'site-1', 'site-2' and 'site-3', the ``robots.txt`` files will
```
Comment (nicolasThal) [inline en/developer_guide/robots.rst]: whose ids are

File: en/developer_guide/robots.rst [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+- web/site-2/robots.txt
+- web/site-3/robots.txt
+
+A good practice is to use that command in a cron, to refresh periodically the content of the files.
+
+.. _files routing:
+
+Files routing
+-------------
+When a client requests a ``robots.txt``, it requests it at the root of the domain. But the files are
```
Comment (nicolasThal) [inline en/developer_guide/robots.rst]: it should be located at the root ...

File: en/developer_guide/robots.rst [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+generated somewhere else, in a subfolder named by the site id. So the webserver must be configured to
```
Comment (nicolasThal) [inline en/developer_guide/robots.rst]: As the files are generated in a subfolder, the webserver must be configured to redirect the request to the matching file.

File: en/developer_guide/robots.rst [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+forward the initial request to the matching file: this can be done by a rewrite rule . The type of
+configuration to tweak depends on you server type, for instance on Apache this is done via the Virtual
```
Comment (nicolasThal) [inline en/developer_guide/robots.rst:61]: We would provide a exemple of configuration for `Apache`_ :

File: en/developer_guide/robots.rst [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+Host mechanism.
+
+Assuming Open Orchestra is running with Apache, and that each site get its own Virtual Host, here what
```
Comment (nicolasThal) [inline en/developer_guide/robots.rst]: You already told that the configuration will be done for open orchestra

File: en/developer_guide/robots.rst [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+could look like a rewrite rule for a site with id 'my-site':
+
+    <IfModule mod_rewrite.c>
+        RewriteEngine On
+        RewriteRule ^/robots.txt /my-site/robots.txt [L]
+    </IfModule>
+
+After an Apache restart, the ``robots.txt`` file for the site 'my-site' is accessible. Note that you can
```
Comment (nicolasThal) [inline en/developer_guide/robots.rst]: Once `Apache2` is reloaded, the rewrite rule will be used and the `robots.txt` accessible.

Every further modification that would be given to the file won't modify the rewrite rule, this way you could update the file periodically

**Comment by ngilain** [comment]
@nicolasThal , @itkg-rboessel , @itkg-alavieille : doc updated

File: en/developer_guide/robots.rst [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+
+Files generation
+----------------
+Once contributed, the content of the ``robots.txt`` is only known from the database. To serve it to
+a client requesting it, two approaches are possible: use a controller action looking in the database
+for the information matching the current site, or serve a previously generated file containing these
+information.
+
+To enhance perfomances, Open Orchestra choose the second option. This solution prevents any PHP action
+and datatbase access. As these files are static they can easily be cached by a reverse proxy.
```
Comment (nicolasThal) [inline en/developer_guide/robots.rst]: static, so they

File: en/developer_guide/sitemap.rst [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+- `files routing`_
+
+.. _files contribution:
+
+Files contribution
+------------------
+Contributing the content of a ``sitemap.xml`` file is done in the Backoffice, and two locations
+are involved: one for the default values, and one for the specific values.
+
+The first place to go is website edition. You can fill in the edit form default values for all pages
```
Comment (nicolasThal) [inline en/developer_guide/sitemap.rst]: In the website edition form, you could set up the default values (see website creation doc for further information).

File: en/developer_guide/sitemap.rst [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+of the site. These settings are the periodicity of change of a page (the changefreq setting) and the
+relative importance of a page compared to the other pages (the priority setting). When set, these
+values are used when a page does not specify its own settings.
+
+But if you want to set a specific setting for a page, you have to edit the matching node setting form.
```
Comment (nicolasThal) [inline en/developer_guide/sitemap.rst]: To set up specific settings for a page, you have ...

File: en/developer_guide/sitemap.rst [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+Here you can set the priority and the changefreq only for this page. The settings given here override
+the website default values.
```
Comment (nicolasThal) [inline en/developer_guide/sitemap.rst]: You could also add a quick link to the node configuration page

File: en/developer_guide/robots.rst [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+
+Creation of ``robots.txt`` files with Open Orchestra is achieved in 3 steps:
+
+- `files contribution`_
+- `files generation`_
+- `files routing`_
+
+.. _files contribution:
+
+Files contribution
```
Comment (alavieille) [inline en/developer_guide/robots.rst:17]: may be a picture to illustrate

**Comment by ngilain** [comment]
@nicolasThal , @itkg-alavieille : update done

File: en/developer_guide/robots.rst (hunk 12 above)
Comment (veolys-nanne) [inline en/developer_guide/robots.rst]: datatbase => database

**Comment by veolys-nanne** [comment]
:+1: after fix

**Pull request merged**